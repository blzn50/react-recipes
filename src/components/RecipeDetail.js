import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const RecipeDetail = ({ recipe, className, ...props }) => {
  if (!recipe) {
    return (
      <div style={props.style} className={classNames('h3 p2 italic bg-white center', className)}>
        <p>Please select a recipe to view detail.</p>
      </div>
    );
  }
  return (
    <div style={props.style} className={classNames('bg-white p2', className)}>
      <h2 className="h2">{recipe.name}</h2>
      <img className="fit" src={recipe.image} alt={recipe.name} />
      <div>
        <span>{recipe.category}</span>
        <span>{recipe.calories}</span>
        <h3>Ingredients</h3>
        {recipe.ingredients && (
          <ul>
            {recipe.ingredients.map(ingredient => (
              <li key={ingredient}>{ingredient}</li>
            ))}
          </ul>
        )}
        <h3>Steps</h3>
        {recipe.steps && (
          <ol>
            {recipe.steps.map(step => (
              <li key={step}>{step}</li>
            ))}
          </ol>
        )}
      </div>
      <Link to={`/recipe/${recipe.id}`}>See More</Link>
    </div>
  );
};

RecipeDetail.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  recipe: PropTypes.object,
};

export default RecipeDetail;
