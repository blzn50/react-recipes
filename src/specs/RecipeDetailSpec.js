import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RecipeDetail from '../components/RecipeDetail';

describe('<RecipeDetail />', () => {
  let testRecipe;
  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'Test recipe',
      category: 'Test category',
      ingredients: ['ing 1', 'ing 2'],
      steps: ['step 1', 'step 2'],
    };
  });

  test('should render without recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should correctly render recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should correctly assign className', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} className="test-classname" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should recipe render without ingredients', () => {
    delete testRecipe.ingredients;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should recipe render without steps', () => {
    delete testRecipe.steps;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
