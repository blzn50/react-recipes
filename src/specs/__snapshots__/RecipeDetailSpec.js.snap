// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> should correctly assign className 1`] = `
<div
  className="bg-white p2 test-classname"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    <ul>
      <li>
        ing 1
      </li>
      <li>
        ing 2
      </li>
    </ul>
    <h3>
      Steps
    </h3>
    <ol>
      <li>
        step 1
      </li>
      <li>
        step 2
      </li>
    </ol>
  </div>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should correctly render recipe 1`] = `
<div
  className="bg-white p2"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    <ul>
      <li>
        ing 1
      </li>
      <li>
        ing 2
      </li>
    </ul>
    <h3>
      Steps
    </h3>
    <ol>
      <li>
        step 1
      </li>
      <li>
        step 2
      </li>
    </ol>
  </div>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should recipe render without ingredients 1`] = `
<div
  className="bg-white p2"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    <h3>
      Steps
    </h3>
    <ol>
      <li>
        step 1
      </li>
      <li>
        step 2
      </li>
    </ol>
  </div>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should recipe render without steps 1`] = `
<div
  className="bg-white p2"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
    <h3>
      Ingredients
    </h3>
    <ul>
      <li>
        ing 1
      </li>
      <li>
        ing 2
      </li>
    </ul>
    <h3>
      Steps
    </h3>
  </div>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should render without recipe 1`] = `
<div
  className="h3 p2 italic bg-white center"
>
  <p>
    Please select a recipe to view detail.
  </p>
</div>
`;
